numba/listobject.py

```diff
@@ -1038,6 +1038,42 @@ def impl(l, item, start=None, end=None):
     return impl
 
 
+@register_jitable
+def equals(this, other):
+    if len(this) != len(other):
+        return False
+    for i in range(len(this)):
+        if this[i] != other[i]:
+            return False
+    else:
+        return True
+
+
+def _equals_helper(this, other, op):
+    if not isinstance(this, types.ListType):
+        return
+    if not isinstance(other, types.ListType):
+        raise TypingError("list can only be compared to list")
+
+    if op == "equals":
+        def impl(this, other):
+            return equals(this, other)
+    elif op == "not_equals":
+        def impl(this, other):
+            return not equals(this, other)
+    return impl
+
+
+@overload(operator.eq)
```

Comment:
Wondering if it's worth doing something like:
```python
@overload(operator.truth)
def ol_truth(val):
    if isinstance(val, types.Boolean):
        def impl(val):
            return val
        return impl

def _equals_helper(this, other, OP):
    if not isinstance(this, types.ListType):
        return
    if not isinstance(other, types.ListType):
        raise TypingError("list can only be compared to list")

    def impl(this, other):
        def equals(this, other):
            if len(this) != len(other):
                return False
            for i in range(len(this)):
                if this[i] != other[i]:
                    return False
            else:
                return True
        return OP(equals(this, other))
    return impl


@overload(operator.eq)
def impl_equals(this, other):
    return _equals_helper(this, other, operator.truth)


@overload(operator.ne)
def impl_not_equals(this, other):
    return _equals_helper(this, other, operator.not_)
```
saves on switching on strings.

Reply:
Yes! 👀